### ISSUE TITLE
Extract examples into separate file

### ISSUE THREAD

**Issue by greystate** [greystate/XSLT-Reference#10]
To make it easier to get user-contributed examples, these should be kept in a separate file.

Examples could just reference the element/function they "belong" to.

<details>
The main XSLT would need to load the `examples.xml` file using the `document()` function and instead of processing child `<example>` elements, it should process something like `$examples/example[@function = name(current())]`.

Sample format:

``` xml
<examples>
   <example function="function-available">...</example>
   <example element="namespace-alias">...</example>
</examples>
```

</details>

**Comment by octoberclub** [comment]
I've done some work to extract the examples into the separate file. I'm struggling with modifying xslt to match the element name from examples.xml to the corresponding parent node in elements.xml. Is there a way to debug the values when transforming  the xslt?

**Comment by greystate** [comment]
Depending on how you're transforming, you could either just output some debugging info, e.g. by adding some extra output, maybe using a separate template `<xsl:template name="debug">` that you can call whenever you want with [call-template](http://pimpmyxslt.com/reference/#call-template).

I had a quick look at your branch and I'll throw a couple of comments in the code - hope that's ok :)

**Comment by octoberclub** [comment]
@greystate Thanks for your XSLT knowledge. I have created a pull request for the change https://github.com/greystate/XSLT-Reference/pull/14

**Comment by greystate** [comment]
Merged with #14
